xarray/tests/test_computation.py

```diff
@@ -818,6 +820,60 @@ def test_vectorize_dask():
     assert_identical(expected, actual)
 
 
+@requires_dask
+def test_vectorize_dask_dtype():
+    # ensure output_dtypes is preserved with vectorize=True
+    # GH4015
+
+    # integer
+    data_array = xr.DataArray([[0, 1, 2], [1, 2, 3]], dims=("x", "y"))
+    expected = xr.DataArray([1, 2], dims=["x"])
+    actual = apply_ufunc(
+        pandas_median,
+        data_array.chunk({"x": 1}),
+        input_core_dims=[["y"]],
+        vectorize=True,
+        dask="parallelized",
+        output_dtypes=[int],
+    )
+    assert_identical(expected, actual)
+    assert expected.dtype == actual.dtype
+
+    # complex
+    data_array = xr.DataArray([[0 + 0j, 1 + 2j, 2 + 1j]], dims=("x", "y"))
+    expected = data_array.copy()
+    actual = apply_ufunc(
+        identity,
+        data_array.chunk({"x": 1}),
+        vectorize=True,
+        dask="parallelized",
+        output_dtypes=[complex],
+    )
+    assert_identical(expected, actual)
+    assert expected.dtype == actual.dtype
+
```

Comment:
Could you add two tests without `output_dtypes` and check the `dtype` of the result; something along the lines of:
```python
data_array = xr.DataArray([[0, 1, 2], [1, 2, 3]], dims=("x", "y"))
    actual = apply_ufunc(
        identity,
        data_array.chunk({"x": 1}),
        dask="parallelized",
    )

    actual = apply_ufunc(
        pandas_median,
        data_array.chunk({"x": 1}),
        input_core_dims=[["y"]],
        vectorize=True,
        dask="parallelized",
    )

```

Reply:
I've added the suggested tests, but the second changes dtype, so it would fail. I just changed the assert. Please let me know, if this is what you had in mind.

Reply:
Ahh, indeed: `pd.Series([1, 2, 3]).median()` returns a float. The idea is to test if it can work without `output_dtypes`, but if `func` changes the dtype it won't work. Sorry my mistake - suggestions below.